myproject/app.py

```diff
@@ -0,0 +1,174 @@
+from flask import Flask, request, jsonify
+from model import db, User
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.security import generate_password_hash, check_password_hash
+from flask_jwt_extended import JWTManager, create_access_token, jwt_required, get_jwt_identity
+from flask_ngrok import run_with_ngrok
+from flask_bcrypt import Bcrypt
+from flask_mail import Mail 
+from flask_mail import Message
+from itsdangerous import URLSafeTimedSerializer
+import string
+import random
+
+
+
+app = Flask(__name__)
+run_with_ngrok(app)
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///User.db'
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+app.config['MAIL_SERVER']='smtp.gmail.com'
+app.config['MAIL_PORT'] = 587
+app.config['MAIL_USERNAME'] = 'jatinmathur1157@gmail.com'
+app.config['MAIL_PASSWORD'] = 'exgxwbiykhostjep'
+app.config['MAIL_USE_TLS'] = True
+app.config['MAIL_USE_SSL'] = False
+db.init_app(app)
+jwt = JWTManager(app)
+bcrypt = Bcrypt(app)
+mail = Mail(app)
+mail.init_app(app)
+
+
+password_reset_tokens = {}
+
+
+@app.route("/register", methods=['POST'])
+def register_user():
+    data = request.json
+    email = data.get('email')
+    first_name = data.get('first_name')
+    last_name = data.get('last_name')
+    phone_no = data.get('phone_no')
+    password = data.get('password')
+    if not all([email, first_name, last_name, phone_no, password]):
+        return jsonify({'error': 'All fields need to be provided'})
+    if User.query.filter_by(email=email).first():
+        return jsonify({'error': 'User with this email already exists'})
+    hashed_password = bcrypt.generate_password_hash(password).decode('utf-8')
+    new_user = User(email=email, first_name=first_name, last_name=last_name,
+                    phone_no=phone_no, password=hashed_password)
+    db.session.add(new_user)
+    db.session.commit()
+    return jsonify({'message': 'User created successfully'})
+
+@app.route('/login', methods=['POST'])
+def login():
+    data = request.get_json()
+    user = User.query.filter_by(phone_no=data['phone_no']).first()
```

Comment:
During the login process, only email and password or username and password are required, and a phone number is not necessary.